src/DynamicData/List/ChangeSet.cs

```diff
@@ -42,145 +35,79 @@ public ChangeSet()
         /// <param name="items">The items.</param>
         /// <exception cref="System.ArgumentNullException">items</exception>
         public ChangeSet([NotNull] IEnumerable<Change<T>> items)
+            : base(items)
         {
-            if (items == null)
-            {
-                throw new ArgumentNullException(nameof(items));
-            }
-
-            var list = items as List<Change<T>> ?? items.ToList();
-
-            Items = list;
-            foreach (var change in list)
-            {
-                Add(change, true);
-            }
         }
 
         /// <summary>
-        /// Adds the specified item.
-        /// </summary>
-        /// <param name="item">The item.</param>
-        public void Add(Change<T> item)
-        {
-            if (item == null)
-            {
-                throw new ArgumentNullException(nameof(item));
-            }
-
-            Add(item, false);
-        }
-
-        /// <summary>
-        /// Adds the specified item.
+        ///     Gets the number of additions
         /// </summary>
-        /// <param name="item">The item.</param>
-        /// <param name="countOnly">set to true if the item has already been added</param>
-        private void Add(Change<T> item, bool countOnly)
+        public int Adds
         {
-            switch (item.Reason)
+            get
             {
-                case ListChangeReason.Add:
-                    _adds++;
-                    break;
-                case ListChangeReason.AddRange:
-                    _adds = _adds + item.Range.Count;
-                    break;
-                case ListChangeReason.Replace:
-                    _replaced++;
-                    break;
-                case ListChangeReason.Remove:
-                    _removes++;
-                    break;
-                case ListChangeReason.RemoveRange:
-                    _removes = _removes + item.Range.Count;
-                    break;
-                case ListChangeReason.Refresh:
-                    _removes = _refreshes++;
```

Comment:
I did notice this potential bug here, where Refresh events were contributing to the remove counter instead of the refresh counter.  This should be implicitly fixed in this PR's refactor.